helm-charts/splunk-otel-collector/templates/operator/_instrumentation-spec.tpl

```diff
@@ -0,0 +1,38 @@
+{{- define "splunk-otel-collector.operator.instrumentation.spec" -}}
+{{- if .Values.operator.enabled }}
+exporter:
+  endpoint: {{- include "splunk-otel-collector.operator.instrumentation.exporter.endpoint" . | nindent 4 }}
+propagators:
+  - tracecontext
+  - baggage
+  - b3
+env:
+  {{- if .Values.agent.enabled }}
+  - name: SPLUNK_OTEL_AGENT
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: status.hostIP
+  {{- end }}
+  {{- if .Values.splunkObservability.profilingEnabled }}
+  - name: SPLUNK_PROFILER_ENABLED
+    value: "true"
+  - name: SPLUNK_PROFILER_MEMORY_ENABLED
+    value: "true"
+  {{- end }}
+python:
+  env:
+    # Required if endpoint is set to 4317.
+    # Python auto-instrumentation uses http/proto by default, so data must be sent to 4318 instead of 4317.
+    # See: https://github.com/open-telemetry/opentelemetry-operator#opentelemetry-auto-instrumentation-injection
+    - name: OTEL_EXPORTER_OTLP_ENDPOINT
+      value: {{- include "splunk-otel-collector.operator.instrumentation.exporter.endpoint" . | replace "4317" "4318" | nindent 6 }}
```

Comment:
Can we maybe change `splunk-otel-collector.operator.instrumentation.exporter.endpoint` to return without the port instead of doing the replace?

Reply:
The suggested change would be harder to support over time because the upstream project uses host + port. I can refactor this code to be more well intentioned, that should help.
- Added a conditional statement to validate ":4317" is at the end of the endpoint value.

Reply:
Done